test/unit/dom/innerHTML.js

```diff
@@ -0,0 +1,231 @@
+          {value: '<!-- comment outside --> text outside <div>text inside</div>', nodes: 3, assigned: 2},
+          {value: 'text outside <!-- comment outside --> text outside <div>text inside <!-- comment inside --> text inside</div>', nodes: 4, assigned: 3},
+        ];
+
+        switch(type) {
+          case 'div':
+            elem = div;
+            break;
+          case 'slot':
+            elem = slot;
+            break;
+          case 'root':
+            root.innerHTML = '';
+            elem = root;
+            break;
+          default:
+            elem = host;
+        }
+      });
+
+      it(`should set and get correct value`, () => {
+        innerHTMLs.forEach(html => {
+          elem.innerHTML = html.value;
+
+          expect(elem.innerHTML).to.equal(html.value);
+          expect(elem.childNodes.length).to.equal(html.nodes);
+
+          if (type !== 'host') {
+            canPatchNativeAccessors && expect(elem.__innerHTML).to.equal(html.value);
+          } else {
+            expect(slot.assignedNodes().length).to.equal(html.assigned);
+          }
+        });
+      });
+
+      it(`should set and get correct value with attributes`, () => {
+        elem.innerHTML = '<div [ foo ] ( bar )></div>';
+        expect(hasAllAttributes(elem.firstChild, ['[', 'foo', ']', '(', 'bar', ')'])).to.equal(true);
+        expect(elem.childNodes.length).to.equal(1);
+
+        if (type === 'host') {
+          expect(slot.assignedNodes().length).to.equal(1);
+        }
+
+        elem.innerHTML = '<div [ foo ] ( bar )><div [ foo2 ] ( bar2 )></div></div>';
+        expect(hasAllAttributes(elem.firstChild, ['[', 'foo', ']', '(', 'bar', ')'])).to.equal(true);
+        expect(hasAllAttributes(elem.firstChild.firstChild, ['[', 'foo2', ']', '(', 'bar2', ')'])).to.equal(true);
+      });
+
+      it('created text nodes get escaped when being appended', function () {
+        const text = document.createTextNode('<u>foo & bar</u>');
+        const textCont = '&lt;u&gt;foo &amp; bar&lt;/u&gt;';
+
+        elem.appendChild(text);
+        expect(elem.innerHTML).to.equal(textCont);
+        expect(elem.childNodes.length).to.equal(1);
+
+        if (type !== 'host') {
+          canPatchNativeAccessors && expect(elem.__innerHTML).to.equal(textCont);
+        } else {
+          expect(slot.assignedNodes().length).to.equal(1);
+        }
+      });
+
+      it('innerHTML handles non text / html / comment nodes', function () {
+        expect(elem.innerHTML).to.equal('');
+        const processingInstruction = '<?xml-stylesheet href="mycss.css" type="text/css"?>';
+        const processingInstructionsAfterInnerHtml = '<!--?xml-stylesheet href="mycss.css" type="text/css"?-->';
+
+        elem.innerHTML = processingInstruction;
+        expect([processingInstruction, processingInstructionsAfterInnerHtml].indexOf(elem.innerHTML)).to.be.above(-1); // different browsers process this differently
```

Comment:
seems a bit risky passing/ok on either value, would be good to explicitly say which browsers don't perform commenting although that might not be feasible